src/components/pages/roster-page/edit-roster-entry-dialog.tsx

```diff
@@ -0,0 +1,174 @@
+    updateRosterEntryProperty(event.target.id, event.target.value);
+  };
+
+  const onCheckboxChanged = (event: React.ChangeEvent<HTMLInputElement>, checked: boolean) => {
+    updateRosterEntryProperty(event.target.id, checked);
+  };
+
+  const onSave = async () => {
+    setFormDisabled(true);
+    try {
+      setSaveRosterEntryLoading(true);
+      if (existingRosterEntry) {
+        await axios.put(`api/roster/${orgId}/${rosterEntry!.edipi}`, rosterEntry);
+      } else {
+        await axios.post(`api/roster/${orgId}`, rosterEntry);
+      }
+    } catch (error) {
+      if (onError) {
+        let message = 'Internal Server Error';
+        if (error.response?.data?.errors && error.response.data.errors.length > 0) {
+          message = error.response.data.errors[0].message;
+        }
+        onError(message);
+      }
+      setFormDisabled(false);
+      return;
+    } finally {
+      setSaveRosterEntryLoading(false);
+    }
+    if (onClose) {
+      onClose();
+    }
+  };
+
+  const canSave = () => {
+    if (formDisabled) {
+      return false;
+    }
+
+    let result = true;
+
+    const requiredColumns = rosterColumnInfos?.filter(columnInfo => columnInfo.required);
+    if (requiredColumns) {
+      for (let i = 0; i < requiredColumns?.length; i++) {
+        const value = rosterEntry[requiredColumns[i].name];
+        if (!value || value.length === 0) {
+          result = false;
+          break;
+        }
+      }
+    }
+
+    return result;
+  };
+
+
+  const buildCheckboxFields = () => {
+    const columns = rosterColumnInfos?.filter(columnInfo => columnInfo.type === 'boolean');
+    return columns?.map(columnInfo => (
+      <TableRow key={columnInfo.name}>
+        <TableCell className={classes.textCell}>
+          {columnInfo.displayName}
+        </TableCell>
+        <TableCell className={classes.iconCell}>
+          <Checkbox
+            color="primary"
+            id={columnInfo.name}
+            disabled={existingRosterEntry ? formDisabled || !columnInfo.updatable : false}
+            checked={rosterEntry[columnInfo.name] || false}
+            onChange={onCheckboxChanged}
+          />
+        </TableCell>
+      </TableRow>
+    ));
+  };
+
+  const buildTextFields = () => {
+    const columns = rosterColumnInfos?.filter(columnInfo => columnInfo.type !== 'boolean');
+    return columns?.map(columnInfo => (
+      <Grid key={columnInfo.name} item xs={6}>
+        <TextField
+          className={classes.textField}
+          id={columnInfo.name}
+          label={columnInfo.displayName}
+          disabled={existingRosterEntry ? formDisabled || !columnInfo.updatable : false}
+          required={columnInfo.required}
+          onChange={onTextFieldChanged}
+          value={columnInfo.type === 'date' && rosterEntry[columnInfo.name] ? rosterEntry[columnInfo.name].split('T')[0] || '' : rosterEntry[columnInfo.name] || ''}
+          type={columnInfo.type === 'date' ? 'date' : 'text'}
```

Comment:
Right now I have text fields to take care of anything that's not a boolean. The type is just different for dates.

1. Would it be worth separating our dates into their own method as well like I did with the booleans? It would clean up lines like the 'value' line here a good amount, but adds another chunk of code. Not sure if it's worth it or not and could use a second opinion on that.

2. Right now the picker only sets/shows the date, not the time. Is it worth using a DateTime Picker for these or is just the date fine? Not sure what's preferred here.